streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -83,6 +84,7 @@ protected int maxNumPartitions(Cluster metadata, Set<String> topics) {
 
             if (partitions == null) {
                 log.info("Skipping assigning topic {} to tasks since its metadata is not available yet", topic);
+                maxNumPartitions = StreamPartitionAssignor.NOT_AVAILABLE;
```

Comment:
@guozhangwang I think, I missed to insert an `break;` here. Otherwise this only works if the "last" topic is not known. Or actually, we can replace this statement with `return StreamPartitionAssignor.NOT_AVAILABLE;`